apps/src/lib/util/audioApi.js

```diff
@@ -109,6 +114,60 @@ export const commands = {
     } else {
       Sounds.getSingleton().stopAllAudio();
     }
+  },
+  playSpeech(opts) {
+    apiValidateType(opts, 'playSpeech', 'text', opts.text, 'string');
+    apiValidateType(opts, 'playSpeech', 'gender', opts.gender, 'string');
+    apiValidateType(
+      opts,
+      'playSpeech',
+      'language',
+      opts.language,
+      'string',
+      OPTIONAL
+    );
+
+    const speechConfig = SpeechConfig.fromAuthorizationToken(
+      appOptions.azureSpeechServiceToken,
+      appOptions.azureSpeechServiceRegion
+    );
+
+    let voice;
+    if (opts.gender === 'male') {
+      voice = 'en-US-BenjaminRUS';
+    } else {
+      voice = 'en-US-AriaRUS';
+    }
+    const synthesizer = new SpeechSynthesizer(speechConfig);
+    let ssml = `<speak version="1.0" xmlns="https://www.w3.org/2001/10/synthesis" xml:lang="en-US"><voice name="${voice}">${
+      opts.text
+    }</voice></speak>`;
+    synthesizer.speakSsmlAsync(
+      ssml,
+      result => {
+        let forceHTML5 = false;
+        if (window.location.protocol === 'file:') {
+          // There is no way to make ajax requests from html on the filesystem.  So
+          // the only way to play sounds is using HTML5. This scenario happens when
+          // students export their apps and run them offline. At this point, their
+          // uploaded sound files are exported as well, which means varnish is not
+          // an issue.
+          forceHTML5 = true;
+        }
+        Sounds.getSingleton().playBytes(result.audioData, {
+          volume: 1.0,
+          loop: !!false,
+          forceHTML5: forceHTML5,
+          allowHTML5Mobile: true
+        });
+
+        synthesizer.close();
+      },
+      error => {
+        console.log(error);
```

Comment:
nit: remove?

Reply:
any recommendations on what I should do with the error? I feel a little weird about letting the text to speech fail silently

Reply:
update: changed it to a console.warn